pkg/dockerregistry/server/blobdescriptorservice.go

```diff
@@ -25,6 +38,155 @@ type blobDescriptorService struct {
 	distribution.BlobDescriptorService
 }
 
+// Stat returns a a blob descriptor if the given blob is either linked in repository or is referenced in
+// corresponding image stream. This method is invoked from inside of upstream's linkedBlobStore. It expects
+// a proper repository object to be set on given context by upper openshift middleware wrappers.
 func (bs *blobDescriptorService) Stat(ctx context.Context, dgst digest.Digest) (distribution.Descriptor, error) {
-	return dockerRegistry.BlobStatter().Stat(ctx, dgst)
+	repo, repoFound := RepositoryFrom(ctx)
+	if !repoFound {
+		context.GetLogger(ctx).Errorf("failed to retrieve repository from context")
+	}
+
+	// if there is a repo layer link, return its descriptor
+	desc, err := bs.BlobDescriptorService.Stat(ctx, dgst)
+	if err == nil {
+		if repoFound {
```

Comment:
The above check doesn't return. It just logs the error. I wanted to fallback to regular stat if the repository isn't set. But I guess that in such a case something went badly wrong and we should return - I'll make it so.